### PR TITLE
Cleanup sc-server round 1

### PR DESCRIPTION
**Pull request by gadfort**
Changes:
- fix sc-configure to allow for port to be entered
- removed calls to mkdir via subprocess and instead use builtin makedirs
- removed unused code from sc-server
- added local server tests to CI and updated to sc-server to function again.

This is a step towards getting all server tests with slurm enabled as well.

### CODE REVIEW

File: siliconcompiler/server.py [hunk 1]
```diff
@@ -156,8 +159,8 @@ async def handle_remote_run(self, request):
         # Create a dummy Chip object to make schema traversal easier.
         # TODO: if this is a dummy Chip we should be able to use Schema class,
         # but looks like it relies on chip.status.
-        chip = Chip(cfg['design']['value'])
-        chip.schema.cfg = cfg
+        chip = Chip('server')  # start with a dummy name, as this will be overwritten
+        chip.schema = Schema(cfg=cfg) # Add provided schema
```
Comment (WRansohoff) [inline siliconcompiler/server.py:163]: This is a very minor comment, but I'm not sure about putting comments at the end of lines in our public code. It's nice for brief examples and documentation, but maybe move these comments to the line above?

File: siliconcompiler/server.py [hunk 2]
```diff
@@ -419,35 +417,40 @@ async def remote_sc_auth(self, chip, username, pk):
...
+            keypath = f'{build_dir}/pk'
             with open(os.open(keypath, os.O_CREAT | os.O_WRONLY, 0o400), 'w+') as keyfile:
                 keyfile.write(pk)
             chip.write_manifest(f"{build_dir}/configs/chip{chip.get('option', 'jobname')}.json")
-            # Create the command to run.
-            run_cmd  = f"cp -R {from_dir}/* {to_dir}/ ; "
+
+            utils.copytree(from_dir, build_dir, dirs_exist_ok=True)
+
+            utils.copytree(from_dir, build_dir, dirs_exist_ok=True)
```
Comment (WRansohoff) [inline siliconcompiler/server.py:441]: Do we need both `copytree` calls?

**Review by WRansohoff — APPROVED** [review]
LGTM, just one possibly-duplicate call that you might want to remove.

File: siliconcompiler/server.py (hunk 2 above)
Comment (gadfort) [reply siliconcompiler/server.py:441]: looks like I was a little trigger happy with the copy and paste.

**Pull request merged**